terraform/groups/configurable-api-caller/data.tf

```diff
@@ -0,0 +1,79 @@
+data "vault_generic_secret" "stack_secrets" {
+  path = "applications/${var.aws_account}/${var.environment}/${local.stack_name}-stack"
+}
+
+data "vault_generic_secret" "service_secrets" {
+  path = "applications/${var.aws_account}/${var.environment}/${local.stack_name}-stack/${var.service}"
+}
+
+data "aws_vpc" "vpc" {
+  filter {
+    name   = "tag:Name"
+    values = [local.vpc_name]
+  }
+}
+
+data "aws_subnets" "application" {
+  filter {
+    name   = "vpc-id"
+    values = [data.aws_vpc.vpc.id]
+  }
+
+  filter {
+    name   = "tag:Name"
+    values = [local.application_subnet_pattern]
+  }
+}
+
+data "aws_iam_policy_document" "get_param_store_systems_manager_policy" {
+  statement {
+    effect  = "Allow"
+    actions = [
+      "ssm:GetParameterHistory",
+      "ssm:GetParametersByPath",
+      "ssm:GetParameters",
+      "ssm:GetParameter"
+    ]
+    resources = ["*"]
```

Comment:
This is too open and should be restricted to the parameter store paths that this service requires only.